functions/src/routes/experiences/experienceHistory.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable promise/no-nesting */
+const admin = require("firebase-admin");
+
+const db = admin.firestore();
+
+const experienceHistory = (req, res) => {
+    const {id: userId} = req.payload;
+    const {type = 'all'} = req.query;
+    let experienceRef = db.collection("experiences");
+
+    if (type === 'current') {
+        experienceRef = experienceRef
+            .where("allAttendees", "array-contains", userId)
+            .where("status", "==", "Open")
+    } else {
+        experienceRef = experienceRef
+            .where("allAttendees", "array-contains", userId)
+    }
+    
+    return experienceRef
+        .get()
+        .then(snapshot => {
+            const data = snapshot.docs.map(snap => {
+                return {...snap.data(), id: snap.id};
+            });
+            return res.status(200).send({
+                success: true,
+                message: "History Retrieved",
+                data
+            });
+        })
```

Comment:
And for this endpoint, please find a way to return just an object if it the type is current else return an array

Reply:
Does that mean you'll be checking for the typeof data on the FrontEnd??, since it'll be returning different types

Reply:
Does that mean you'll be checking for the typeof data on the FrontEnd??, since it'll be returning different types

Reply:
So long as I send "current" as the type, I know I will get back an object

Reply:
Alright cool